### PR TITLE
Update AWS Executor documentation

### PR DESCRIPTION
**Pull request by mahammi**
A quick fix to AWS ECS and Batch executor docs to link to the correct path for the Dockerfile given to users. 

As well, I have added an extra instruction for building docker images for those running Apple Silicon. In order for the ECS Executor to run successfully, the user must specify x86 architecture. This can be done by building the image using`docker buildx`, and specifying arch using the `--platform` flag.

### CODE REVIEW

File: docs/apache-airflow-providers-amazon/executors/ecs-executor.rst [hunk 1]
```diff
@@ -187,6 +187,13 @@ Then you can build your image by ``cd``-ing to the directory with the Dockerfile
    docker build -t my-airflow-image \
     --build-arg aws_default_region=YOUR_DEFAULT_REGION .
 
+For users on Apple Silicon, you must specify the arch using ``docker buildx``:
```
Comment (vincbeck) [inline docs/apache-airflow-providers-amazon/executors/ecs-executor.rst]: Could you please update batch executor documentation as well. It is done here: https://github.com/apache/airflow/blob/main/docs/apache-airflow-providers-amazon/executors/general.rst?plain=1#L68. I wondered if we should not use this one in ECS executor as well? I guess that's why we moved this section to a separate file? @syedahsn @o-nikolas

Comment (vincbeck) [reply docs/apache-airflow-providers-amazon/executors/ecs-executor.rst]: But thanks for adding this section! :D Very helpful for Mac users

Comment (o-nikolas) [reply docs/apache-airflow-providers-amazon/executors/ecs-executor.rst]: I was just digging all this up and got distracted before finishing it. Thanks @vincbeck!

I'm not sure why we haven't circled back to update ECS to use the generic doc imports. @syedahsn added them during the Batch project and perhaps we just haven't circled back?

Comment (vincbeck) [reply docs/apache-airflow-providers-amazon/executors/ecs-executor.rst]: Let's do it then! :)

Comment (syedahsn) [reply docs/apache-airflow-providers-amazon/executors/ecs-executor.rst]: Yes, I'm pretty sure it was something that was on my todo list, but never got around to it. The ECS guide can be re-done to use the general document, similar to the Batch documentation.

Comment (mahammi) [reply docs/apache-airflow-providers-amazon/executors/ecs-executor.rst]: Thanks for clarifying everyone :) I modified ECS docs to now make use of `general.rst`, and modified `general.rst` to include Mac build instructions!

Comment (Taragolis) [reply docs/apache-airflow-providers-amazon/executors/ecs-executor.rst]: It is not 100% correct, e.g. if it intend to be run under [Graviton](https://aws.amazon.com/ec2/graviton/) then it should be `--platform=linux/arm64`.

BTW It is better to build on the same architecture as on which intended to be launched

Comment (Taragolis) [reply docs/apache-airflow-providers-amazon/executors/ecs-executor.rst]: So maybe instead of cover one case, better cover most of them with mention that is important to build and run under the same architecture. And also would be nice to have a link to the Docker documentation.

And keep example with a small clarification, that if you build on Apple Silicone and intend to run on X86_64 aka AMD64 then make sure that you use appropriate platform.

File: docs/apache-airflow-providers-amazon/executors/ecs-executor.rst [hunk 2]
```diff
@@ -19,6 +19,8 @@
 .. warning::
    The ECS Executor is alpha/experimental at the moment and may be subject to change without warning.
 
+.. |executorName| replace:: ECS
+.. |dockerfileLink| replace:: `here <https://github.com/apache/airflow/blob/main/airflow/providers/amazon/aws/executors/Dockerfile>`__
```
Comment (vincbeck) [inline docs/apache-airflow-providers-amazon/executors/ecs-executor.rst:23]: Do we need to set `configKwargs`? `general.rst` uses it

File: docs/apache-airflow-providers-amazon/executors/ecs-executor.rst (hunk 1 above)
Comment (vincbeck) [reply docs/apache-airflow-providers-amazon/executors/ecs-executor.rst]: Valid points @Taragolis, I think the guide assumes that you are building the image on the same machine you are running it but I guess you are right we can add minor modifications to make it clear.

> So maybe instead of cover one case, better cover most of them with mention that is important to build and run under the same architecture.

Covering all cases can be cumbersome (and especially, what are all cases?). We could mention "that it is important to build and run under the same architecture" and give as example the Apple Silicon

> And also would be nice to have a link to the Docker documentation.

Agree

File: docs/apache-airflow-providers-amazon/executors/general.rst [hunk 3]
```diff
@@ -74,6 +74,13 @@ Then you can build your image by ``cd``-ing to the directory with the Dockerfile
    docker build -t my-airflow-image \
     --build-arg aws_default_region=YOUR_DEFAULT_REGION .
 
+For users on Apple Silicon, you must specify the arch using ``docker buildx``:
+
+.. code-block:: bash
+
+  docker buildx build --platform=linux/amd64 -t my-airflow-image \
```
Comment (jedcunningham) [inline docs/apache-airflow-providers-amazon/executors/general.rst:82]: (driveby comment) It might be worth adding the platform in the dockerfile instead. Less likely to get forgotten that way.

```
FROM --platform=linux/amd64 someimage
```

File: docs/apache-airflow-providers-amazon/executors/ecs-executor.rst (hunk 1 above)
Comment (mahammi) [reply docs/apache-airflow-providers-amazon/executors/ecs-executor.rst]: Sorry I forgot to follow up on this!

I've now added a general comment noting that we must build and run under the same architecture. I added the mac example command, and also linked Docker documentation.

File: docs/apache-airflow-providers-amazon/executors/ecs-executor.rst (hunk 2 above)
Comment (mahammi) [reply docs/apache-airflow-providers-amazon/executors/ecs-executor.rst:23]: You're right! done!

File: docs/apache-airflow-providers-amazon/executors/general.rst (hunk 3 above)
Comment (mahammi) [reply docs/apache-airflow-providers-amazon/executors/general.rst:82]: I'm wondering whether hardcoding the platform in the Dockerfile would cause problems or confusion for the user, if they didn't mean to build _and_  run under x86. 

I think it would be a better idea to leave this up to the user, should they decide they need to specify x86 architeture. What do you think?

**Review by vincbeck — APPROVED** [review]

**Comment by vincbeck** [comment]
Thanks for the changes! LGTM!

File: docs/apache-airflow-providers-amazon/executors/general.rst (hunk 3 above)
Comment (mahammi) [reply docs/apache-airflow-providers-amazon/executors/general.rst:82]: Any thoughts @jedcunningham ? :)

Comment (jedcunningham) [reply docs/apache-airflow-providers-amazon/executors/general.rst:82]: Sorry, I don't have enough context to really have an opinion. My only point was that, if the platform is consistent, doing it in the dockerfile is less error prone than an extra arg in `docker build`. If the platform does vary, the extra arg is the way to go. Let's just say I have first hand knowledge that it gets missed sometimes :)

Comment (vincbeck) [reply docs/apache-airflow-providers-amazon/executors/general.rst:82]: Yeah I think have it as a parameter makes more sense because the platform can change depending on the user. This is actually the goal of adding this paragraph in the documentation
